main/auth/auth.py

```diff
@@ -132,6 +132,24 @@ def decorated_function(*args, **kwargs):
   return decorated_function
 
 
+def cron_required(f):
+  decorator_order_guard(f, 'auth.cron_required')
+
+  @functools.wraps(f)
+  def decorated_function(*args, **kwargs):
+    if config.PRODUCTION and 'X-Appengine-Cron' in flask.request.headers:
+      return f(*args, **kwargs)
+    if is_logged_in() and current_user_db().admin:
+      return f(*args, **kwargs)
+    if not is_logged_in() and flask.request.path.startswith('/api/'):
+      return flask.abort(401)
+    if not is_logged_in():
+      return flask.redirect(flask.url_for('signin', next=flask.request.url))
+    return flask.abort(403)
+
+  return decorated_function
+
+
```

Comment:
Understand, neither did I; however I tested some more and this works locally on the devserver:

```
def cron_required(f):
  decorator_order_guard(f, 'auth.cron_required')

  @functools.wraps(f)
  def decorated_function(*args, **kwargs):
    if 'X-Appengine-Cron' in flask.request.headers:
      ip = flask.request.remote_addr
      if ip == '0.1.0.1':
        return f(*args, **kwargs)
    return flask.abort(403)

  return decorated_function
```

When associated with an endpoint, the devserver shows them as Cron Jobs and you can issue a Run Now; and then the unique header/ip combo is like you would see on production.